### ISSUE TITLE
Invalid rule Index for application_error when a module is missing

### ISSUE THREAD

**Issue by tbutler-qontigo** [terraform-linters/tflint#2367]
### Summary

If a module references a missing module then the `results` array includes a `result` object with an invalid `ruleIndex` - e.g.
```json
      "results": [
        {
          "ruleId": "application_error",
          "ruleIndex": 18446744073709551615,
          "level": "error",
          "message": {
            "text": "Failed to run in .; exit status 1\n\nFailed to load configurations; main.tf:6,1-17: \"missing\" module is not found; The module directory \"MissingFolder\" does not exist or cannot be read.:\n\n\u001b[31mError\u001b[0m: \"missing\" module is not found\n\n  on main.tf line 6, in module \"missing\":\n   6: \u001b[1;4mmodule \"missing\"\u001b[0m {\n\nThe module directory \"MissingFolder\" does not exist or cannot be read.\n\n"
          }
        }
      ]
```
According to the SARIF spec [here](https://docs.oasis-open.org/sarif/sarif/v2.1.0/errata01/os/sarif-v2.1.0-errata01-os-complete.html#_Toc141790894) the `ruleIndex` should either be an array index OR should be absent from the results.

Reproducer attached - unzip, move to the folder, then:
```
tflint --init
tflint --recursive --force --format=sarif 
```

This produces output as per attached:
[tflint.json](https://github.com/user-attachments/files/22004110/tflint.json)

[Reproducer](https://github.com/user-attachments/files/22004150/TestSARIF.zip)

### Command

tflint --recursive --force --format=sarif

### Terraform Configuration

```terraform
See attached zip
```

### TFLint Configuration

```HCL
plugin "terraform" {
  enabled = true
  preset  = "all"
}

plugin "azurerm" {
  enabled = true
  version = "0.28.0"
  source  = "github.com/terraform-linters/tflint-ruleset-azurerm"
}
```

### Output

```Shell
{
  "version": "2.1.0",
  "$schema": "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/main/sarif-2.1/schema/sarif-schema-2.1.0.json",
  "runs": [
    {
      "tool": {
        "driver": {
          "informationUri": "https://github.com/terraform-linters/tflint",
          "name": "tflint",
          "rules": [],
          "version": "0.58.0"
        }
      },
      "results": []
    },
    {
      "tool": {
        "driver": {
          "informationUri": "https://github.com/terraform-linters/tflint",
          "name": "tflint-errors",
          "rules": [],
          "version": "0.58.0"
        }
      },
      "results": [
        {
          "ruleId": "application_error",
          "ruleIndex": 18446744073709551615,
          "level": "error",
          "message": {
            "text": "Failed to run in .; exit status 1\n\nFailed to load configurations; main.tf:6,1-17: \"missing\" module is not found; The module directory \"MissingFolder\" does not exist or cannot be read.:\n\n\u001b[31mError\u001b[0m: \"missing\" module is not found\n\n  on main.tf line 6, in module \"missing\":\n   6: \u001b[1;4mmodule \"missing\"\u001b[0m {\n\nThe module directory \"MissingFolder\" does not exist or cannot be read.\n\n"
          }
        }
      ]
    }
  ]
}
```

### TFLint Version

0.58.0

### Operating System

- [ ] Linux
- [ ] macOS
- [x] Windows
